packages/app/src/background/services/validation/source.ts

```diff
@@ -1,51 +1,45 @@
-import { MerkleProofSource, MerkleProofStorageUrl } from "@cryptkeeperzk/types/dist/src/proof/zkProof";
-
-import type { IMerkleProof, IMerkleProofArtifacts, IMerkleProofInputs } from "@cryptkeeperzk/types";
+import { IMerkleProof, IMerkleProofArtifacts, IMerkleProofInputs, MerkleProofStorageUrl } from "@cryptkeeperzk/types";
+import * as yup from "yup";
 
 import { MerkleProofValidator } from "./merkle";
 
+const merkleProofStorageUrlSchema: yup.Schema<MerkleProofStorageUrl> = yup.string().url().required();
+
+const merkleProofArtifactsSchema: yup.Schema<IMerkleProofArtifacts> = yup.object({
+  leaves: yup.array().of(yup.string().required()).required(),
+  depth: yup.number().required(),
+  leavesPerNode: yup.number().required(),
+});
+
+const merkleProofSchema: yup.Schema<IMerkleProof> = yup.object({
+  root: yup.string().required(),
+  leaf: yup.string().required(),
+  siblings: yup.array().of(yup.string().required()).required(),
+  pathIndices: yup.array().of(yup.number().required()).required(),
+});
+
 export const validateMerkleProofSource = ({
   merkleProofSource,
 }: Partial<IMerkleProofInputs>): Partial<IMerkleProofInputs> => {
   if (!merkleProofSource) {
     throw new Error("CryptKeeper: please set a merkle proof source.");
   }
 
-  if (isMerkleProofStorageUrl(merkleProofSource)) {
-    const merkleStorageUrl: MerkleProofStorageUrl = merkleProofSource as MerkleProofStorageUrl;
+  if (typeof merkleProofSource === "string" && merkleProofStorageUrlSchema.isValidSync(merkleProofSource)) {
```

Comment:
Why do you need typeof checks for ifs here?

Reply:
To switch between the types of the inputs, if its string that means most probably a URL. It might be a string object but it would fail.